container/kvm/cache.go

```diff
@@ -0,0 +1,362 @@
+package kvm
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	yaml "gopkg.in/yaml.v1"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/environs/imagedownloads"
+	"github.com/juju/juju/environs/simplestreams"
+	"github.com/juju/juju/juju/paths"
+	"github.com/juju/utils/series"
+)
+
+const (
+	// FType is the file type we want to fetch and use for kvm instances.
+	FType = "disk1.img"
+)
+
+// Oner is an interface which allows us to use sync params to call
+// imagedownloads.One or pass in a fake for testing.
+type Oner interface {
+	One() (*imagedownloads.Metadata, error)
```

Comment:
I plan to update the comments in the later passes, since they change a lot during development.